### PR TITLE
Asher011/mfsim target

### PR DESCRIPTION
**Pull request by AlejandroSherman**
Changes to:
Mfsim_target.py - various additions such bugs fixed to allow for the existing assays to produce synthesize-able output, split semantic rework, conditional group generation additions, live dropplet tracking etc
test_mfsim.py - expanded test suite with existing assay tests and more complex tests, and added another parameter to get_compiled_mfsim in front_end_base.py
Updated existing assays to allow for currently used syntax
Created test files used in test_mfsim

Potential remaining changes mfsim_target may need:
Certain combinations of conditionals in complex assays may need more logic in conditional group generation,
while loops work, though may also need further logic as well for more complex edge cases.

For the while and nested conditional cases I was able to find and fix, each tested case has a working unit test in the test_mfsim test suite, and can be expanded later if others are found to have differing behavior.

### CODE REVIEW

File: tests/test_cases/split/simple_size2.bs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+module fluorescence
+
+instructions:
+
+sa = dispense 10 units of substance_a
+d1 = dispense 1 units of dilutant1
+
+first_dilute = mix sa with d1 for 5s
+x = split first_dilute into 2
+dispose x[0]
```
Comment (tysolov) [inline tests/test_cases/split/simple_size2.bs]: If you are disposing (or performing an identical operation) on all droplets resulting from a split, you should us a SIMD instruction:
example: 
x = split first_dilute into 2
dispose x // this instruction disposes both droplets.

**Review by tysolov — CHANGES_REQUESTED** [review]
use SIMD disposals on tests where appropriate.
remove old/commented code from mfsim_target

update test files with conditionals to include parentheses around the conditional

**Review by tysolov — APPROVED** [review]

**Review by tysolov — CHANGES_REQUESTED** [review]
Everything looks great -- please resolve merge conflicts with main branch and re-request review so we can integrate this.

**Review by tysolov — APPROVED** [review]
